easyaudit/middleware/easyaudit.py

```diff
@@ -37,12 +38,9 @@ def clear_request():
         del _thread_locals.request
 
 
-class EasyAuditMiddleware:
+class EasyAuditMiddleware(MiddlewareMixin):
```

Comment:
You should not convert this to an old-style mixin.

Reply:
`EasyAuditMiddleware` already uses old pre-Django-1.10 convention. `MiddlewareMixin` helps achieve sync/async capability without much boilerplate code.

Would you be ok with going fully with new style middleware?

```suggestion
class EasyAuditMiddleware:
    """Makes request available to this app signals."""
    async_capable = True
    sync_capable = True

    def __init__(self, get_response):
        self.get_response = get_response
        if iscoroutinefunction(self.get_response):
            markcoroutinefunction(self)

    def __call__(self, request):
        if iscoroutinefunction(self):
            return self.__acall__(request)

        _thread_locals.request = request
        response = self.get_response(request)

        with contextlib.suppress(AttributeError):
            del _thread_locals.request

        return response

    async def __acall__(self, request):
        _thread_locals.request = request

        response = await self.get_response(request)

        with contextlib.suppress(AttributeError):
            del _thread_locals.request

        return response
```

Reply:
Yep, that'd be ideal IMO. Get rid of legacy conventions.

Also, don't forget tests!

Reply:
@samamorgan I've pushed requested changes.